src/backend/marsha/core/views.py

```diff
@@ -81,72 +81,175 @@ def exception_handler(exc, context):
     return drf_exception_handler(exc, context)
 
 
-def _get_base_app_data():
-    """Define common app data."""
-    return {
-        "environment": settings.ENVIRONMENT,
-        "flags": {
-            SENTRY: switch_is_active(SENTRY),
-            BBB: settings.BBB_ENABLED,
-            LIVE_RAW: settings.LIVE_RAW_ENABLED,
-            MARKDOWN: settings.MARKDOWN_ENABLED,
-        },
-        "release": settings.RELEASE,
-        "sentry_dsn": settings.SENTRY_DSN,
-        "static": {
-            "svg": {
-                "icons": static("svg/icons.svg"),
+class MarshaViewMixin:
+    """
+    Mixin which provides the very base context data for Marsha's frontend views.
+
+    The context provides information for the `core/site.html` and `core/resource.html`
+    templates:
+     - static_base_url: the base URL for static files (including the React frontend)
+     - external_javascript_scripts: third party js scripts to load
+     - app_data: a payload to provide data to the frontend application
+
+    Views must define the `frontend_name`, for now, only two values: `Site` and `LTI`.
```

Comment:
What do you think about raising an exception if this parameter is not defined ?

Reply:
It may be overkill, actually, if one doesn't define it, it will result in `frontend` app data provided as None, I guess if you need it you will probably quickly provide it ^^ And if you don't need it then it is okay to leave it at `None`. 
I may update the docstring to say "Views should define the [...]". What do you think ?

Reply:
Ok you can change with your suggestion